presto-common/src/main/java/com/facebook/presto/common/block/VariableWidthBlock.java

```diff
@@ -164,7 +164,15 @@ public long getRetainedSizeInBytes()
     @Override
     public void retainedBytesForEachPart(ObjLongConsumer<Object> consumer)
     {
-        consumer.accept(slice, slice.getRetainedSize());
+        // Slices with shared base objects, common scenario when a block is deserialized
```

Comment:
I would reword this a bit.  Explain the problem (for deserialized pages we can have many views on top of an underlying shared Slice object), how this solves it (by slicing up the view reference from the underlying shared byte array, which will not be double counted thanks to reference counting).